core/src/main/java/org/openconnectors/connect/Connector.java

```diff
@@ -54,21 +45,9 @@ public void initialize(final ConnectorContext ctx){
     public abstract void open(final Config config) throws Exception;
 
     /**
-     * Reset the connector with a new configuration
+     * Close the connector
      *
-     * @param config
      * @throws Exception
      */
-    public void reset(final Config config) throws Exception{
-        close();
-        open(config);
-    }
-
-    public ConnectorContext getContext() {
```

Comment:
we still need access to the context

Reply:
Who needs access to the context? If a connectors wants any context, its free to store the pointer during initialize?

Reply:
the adapter layer will run it , say the heron adapter layer or kafka connect layer

Reply:
Here's my bone wrt having something like getContext:- This is a system generated Context. Thus you can just using it to init some stuff(infact I'm thinking maybe an initialize is also not necessary in the Connector interface). The system should pass it to other components if they need it. The connector should not be burdened with it

Reply:
ok